src/renderer/contexts/NodeDocumentationContext.tsx

```diff
@@ -25,11 +33,53 @@ export const NodeDocumentationProvider = memo(({ children }: React.PropsWithChil
         onOpen();
     };
 
+    const { schemata } = useContext(BackendContext);
+    const schema = schemata.schemata;
+    const searchableSchema = useMemo(() => {
+        return schema.map((s) => ({
+            ...s,
+            inputs: s.inputs.map((i) => `${i.label} ${i.description ?? ''}`),
+            outputs: s.outputs.map((o) => `${o.label} ${o.description ?? ''}`),
+        }));
+    }, [schema]);
+
+    const [searchQuery, setSearchQuery] = useState('');
+
+    const idField: keyof NodeSchema = 'schemaId';
+    const fields: (keyof NodeSchema)[] = [
+        'category',
+        'description',
+        'name',
+        'subcategory',
+        'inputs',
+        'outputs',
+    ];
+    const miniSearch = new MiniSearch({ idField, fields });
+
+    miniSearch.addAll(searchableSchema);
+
+    const searchResult = miniSearch.search(searchQuery, {
+        boost: { name: 2 },
+        fuzzy: 0.2,
+        prefix: true,
+        combineWith: 'AND',
+    });
+
+    const searchTerms = searchResult.find((s) => s.id === selectedSchemaId)?.terms ?? [];
+
+    const nodeDocsSearchState = {
+        searchQuery,
+        setSearchQuery,
+        searchResult,
+        searchTerms,
+    };
+
     const contextValue = useMemoObject({
         selectedSchemaId,
         isOpen,
         openNodeDocumentation,
         onClose,
+        nodeDocsSearchState,
```

Comment:
You effectively disabled memoing because `nodeDocsSearchState` is a new object every time.